datamonster_api/lib/aggregation.py

```diff
@@ -1,47 +1,61 @@
-from collections import namedtuple
-
 from .company import Company
 from .errors import DataMonsterError
 
 
-aggregation_periods = {"week", "month", "quarter", "fiscalQuarter", "year"}
-Aggregation = namedtuple("Aggregation", ["period", "company"])
+class Aggregation(object):
+    """A representation of an aggregation type within DataMonster"""
 
-Aggregation.__doc__ = """
-A representation of an aggregation type within DataMonster
+    aggregation_period_map = {
+        "fiscalQuarter": "fiscal quarterly",
+        "quarter": "calendar quarterly",
+        "month": "monthly",
+        "week": "weekly",
+    }
 
-:param str period: The period of time over which to aggregate. Valid choices are:
-    {}
-:param company: The ``Company`` object for which fiscal quarters are calculated.
-    Only relevant for the `fiscalQuarter` aggregation period.
-""".format(
-    aggregation_periods
-)
+    def __init__(self, period, company):
+        """Initialize an Aggregation
 
+        :param str period: The period of time over which to aggregate. Valid choices are:
+            "week", "month", "quarter", "fiscalQuarter", "year"
+        :param company: The ``Company`` object for which fiscal quarters are calculated.
+            Only relevant for the `fiscalQuarter` aggregation period.
+        """
 
-def aggregation_sanity_check(aggregation, company=None):
-    if aggregation.period is not None and aggregation.period not in aggregation_periods:
-        raise DataMonsterError(
-            "Bad Aggregation Period: {}. Valid choices are: {}".format(
-                aggregation.period, aggregation_periods
-            )
-        )
+        self.period = period
+        self.company = company
+        self.aggregation_sanity_check()
 
-    if aggregation.company is not None and not isinstance(aggregation.company, Company):
-        raise DataMonsterError(
-            "Aggregation company must be Company. Found : {}".format(
-                aggregation.company.__class__
-            )
-        )
+    def to_time_aggregation_dictionary(self, aggregation_type='sum'):
+        """Used to extract the time aggregation dictionary for API usage from the Aggregation"""
+        return {
+            'cadence': self.aggregation_period_map[self.period],
+            'aggregationType': aggregation_type,
+            'includePTD': False
```

Comment:
```        
base =  {
            'cadence': self.aggregation_period_map[self.period],
            'aggregationType': aggregation_type,
            'includePTD': False}
        if self.company and self.period == 'fiscalQuarter':
            base['sectionPk']  = self.company.id
```
basically that, since the API doesn't like section being there unless it's fiscal quarterly.